File: from-0601-to-0700/0633-sum-of-square-numbers/solution.go
func judgeSquareSum(c int) bool {
    left := 0
    right := int(math.Floor(math.Sqrt(float64(c))))
    var res int

    for left <= right {
        res = left * left + right * right
        if res == c {
            return true
        } else if res < c {
            left += 1
        } else {
            right -= 1
        }
    }
    return false
}

File: from-0601-to-0700/0633-sum-of-square-numbers/solution.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        left = 0
        right = int(c ** 0.5)
        while left <= right:
            res = left*left + right*right
            if res == c:
                return True
            if res < c:
                left += 1
            else:
                right -= 1

        return False